contracts/contracts/vault/VaultAdmin.sol

```diff
@@ -184,8 +194,29 @@ contract VaultAdmin is VaultStorage {
      * @param _addr Address of the strategy to add
      */
     function approveStrategy(address _addr) external onlyGovernor {
+        _approveStrategy(_addr, false);
+    }
+
+    /**
+     * @dev Add a strategy to the Vault and mark it as Uniswap V3 strategy
+     * @param _addr Address of the strategy to add
+     */
+    function approveUniswapV3Strategy(address _addr) external onlyGovernor {
+        _approveStrategy(_addr, true);
+    }
+
+    /**
+     * @dev Add a strategy to the Vault
+     * @param _addr Address of the strategy to add
+     * @param isUniswapV3 Set to true, if the strategy is an instance of GeneralizedUniswapV3Strategy
+     */
+    function _approveStrategy(address _addr, bool isUniswapV3) internal {
         require(!strategies[_addr].isSupported, "Strategy already approved");
-        strategies[_addr] = Strategy({ isSupported: true, _deprecated: 0 });
+        strategies[_addr] = Strategy({
```

Comment:
Leaving a note here: Strategy structs are mapped via:
```
mapping(address => Strategy) internal strategies
```

the [bool, uint256, bool] cause the Strategy structure to take up 3 storage slots. The new variable `isUniswapV3Strategy` has correctly been added to the end of the struct, ensuring that no old mappings are affected. 

nitpick: in theory because of a struct taking up 3 storage slots there could be collisions but it is really really highly unlikely.